### DISCUSSION TITLE
Tasmota WT32-ETH01 does not start when the LAN cable is disconnected.

### DISCUSSION THREAD

**Discussion by Vitek-117** [arendst/Tasmota#21618] [Support request] [answered]
Good afternoon. I have a problem with the WT32-ETH01 controller(LAN8720). If the LAN cable is disconnected before power is applied. And to supply power to the device, the Tasmota firmware does not start. Until the LAN cable is connected to the WT32-ETH01. After connecting the cable, the firmware immediately starts and loads. Moreover, the problem is not in the specific firmware. And in all tasmota32 firmware.
He writes this in the log:
00:00:00.002 HDW: ESP32-D0WD v1.0
00:00:00.020 UFS: FlashFS mounted with 304 kB free
00:00:00.036 CFG: Loaded from File, Counting 25
00:00:00.046 QPC: Count 1
00:00:00.052 I2C: Bus1 using GPIO32(SCL) and GPIO33(SDA)
00:00:00.090 BRY: Berry initialized, RAM used 3904 bytes
00:00:00.102 DSP: TM1637 with 4 digits (type 0)
00:00:00.112 tasmota Project - Tasmota Version 14.1.0(tasmota32)-3_0_0(2024-06-12T13:05:44)
00:00:04.198 I2C: BMP180 found at 0x77
00:00:11.253 QPC: Reset
And that's it, the firmware does not load further.... Connecting the LAN cable:
00:02:08.103 ETH: Connected at 100Mbps Full Duplex, Mac C8:F0:9E:DE:28:DB, Hostname tasmota-DE28D8-2264-eth
00:02:09.663 HTP: The web server is active tasmota-DE28D8-2264-eth with the IP address 192.168.1.127
00:02:10.661 RSL: INFO1 = {"Info1":{"Module":"WT32-Eth01","Version":"14.1.0(tasmota32)","FallbackTopic":"cmnd/DVES_DE28D8_fb/","GroupTopic":"cmnd/tasmotas/"}}
00:02:10.675 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Ethernet":{"Hostname":"tasmota-DE28D8-2264-eth","IPAddress":"192.168.1.127"}}

How do I fix this firmware bug? Because when power is applied with the LAN cable disconnected, the device turns into a brick. Without a local network, the device DOES NOT WORK AT ALL.
What should I do?
![Screenshot_14](https://github.com/arendst/Tasmota/assets/133482412/6e327ff1-803b-44ee-b2f9-574ab904dbe3)
![Screenshot_8](https://github.com/arendst/Tasmota/assets/133482412/fd4fb4ea-4b20-499e-8026-4c6f27ea4e2a)

**Comment by s-hadinger** [comment]
Most probable cause is that the power supply is too weak for wifi to start  and causes a crash. Wifi start is the moment when the chip draws this most power. 

What exactly happens after 'QPC: Reset"?  Is it still accepting commands via serial or does it reboot?

**Reply by Vitek-117** [reply]
The power supply is good. There are no nutritional failures. The Wi-Fi does not work, as I immediately disabled it with the wifi 0 command.
After "QPC: reset" nothing happens. It just hangs in place and that's it. For a better understanding, I can make a video and post it on YouTube.

**Reply by s-hadinger** [reply]
Si you want it to work without LAN and without ETH, right? 

This is an uncommon setup but nothing wrong. I will test on another device in the following days. 

Anything else you changed in the configuration ? Template? Self compile ?

**Reply by Vitek-117** [reply]
Yes, you understood me correctly. I want it to be able to work without a connected LAN cable and ETH. When necessary, then I will connect the LAN cable. 
I Edited the my_user_config file.h by editing these lines:
#define USE_ETHERNET                             // Add support for ethernet (+20k code)
  #define USE_WT32_ETH01                         // Add support for Wireless-Tag WT32-ETH01
  #define ETH_TYPE          0                    // [EthType] 0 = LAN8720, 1 = TLK110/IP101, 2 = RTL8201, 3 =  #define ETH_ADDRESS       1                    // [EthAddress] 0 = PHY0 .. 31 = PHY31
  #define ETH_CLKMODE       3 
[my_user_config.zip](https://github.com/user-attachments/files/15822036/my_user_config.zip)
![Screenshot_9](https://github.com/arendst/Tasmota/assets/133482412/06ca8ba7-a9d0-49ea-b8e3-72743c39e3e1)

And compiled the tasmota32.factory firmware with the addition of Ethernet support. I can upload the my_user_config file.h Look at what I edited there.

**Reply by s-hadinger** [reply]
ESP32 standard firmware already inludes ETH support. Could you please try with the default firmware?

**Reply by Vitek-117** [reply]
OK. Now I'll try to download the standard firmware.

**Reply by Vitek-117** [reply]
I have now tried the standard firmware. To make the ethernet work, I entered the console
backlog template  {"NAME":"WT32-ETH01","GPIO":[1,1,1,1,1,1,0,0,1,0,1,1,3840,576,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,1,1,0,1,1,0,0,1],"FLAG":0,"BASE":1}; EthAddress 1; EthType 0; EthClockMode 0; module 0; 
And nothing has changed. The result is the same.  The firmware does not boot without a cable. I also turned off Wi-Fi completely with the wifi 0 command

**Comment by Jason2866** [comment]
My WT32-ETH01 does work whether the LAN cable is connected or not when powering the device.
The device is Very! picky with power supply.

**Reply by Vitek-117** [reply]
The power supply is good. There are no nutritional failures. The reason is something else. For a better understanding, I can make a video and post it on YouTube.

**Comment by sfromis** [comment]
When booting another Ethernet-capable device (in this case a Zigbee bridge) with Wifi disabled, and disconnecting Ethernet, on serial I see a similar pattern with Tasmota booting until the `QPC: Reset` point, without further messages (except due to log level). Basically the same as what's posted here.

However, that does by far not mean that the device is "bricked" or "dead", just that without any network available, it can't do much to talk to the world. And without time of day available, other functions may be restricted. Still, the specific device still starts the normal Zigbee coordinator function, meaning that Tasmota is till functional, just within the severe limitations of a broken setup.

**Reply by Vitek-117** [reply]
I have a thermometer with a DS18B20 sensor and temperature output from the sensor to the TM1637 display on the WT32-ETH01 for the test.
So, with Wi-Fi turned off, and a LAN cable. The temperature is not displayed on the display. That is, the device does not boot and does not work.

**Reply by Jason2866** [reply]
Which GPIOs are you using to connect the TM1637 and DS18B20 to the WT32-ETH01?

**Reply by sfromis** [reply]
That does not mean that Tasmota did not boot. As I clearly mentioned, in such a broken setup, functions may be restricted. Still, on my device, creating a simple rule to get a sensor value every TelePeriod worked, and "tickle" a led to indicate that Tasmota was working.

Thus, it is not a good conclusion to make the broad claim of "the device does not boot and does not work", when booting without network available. It may be the case that the way you implemented your setup for passing sensor values to the display is hit by the restrictive environment you caged Tasmota into. Details matter instead of falling back on "does not work", without any information on the specifics.

In my case, while I got it working, depending on the telemetry period did have the effect of Tasmota not being able to do good timing without a real time available. Hence, I did "push" it a bit by just running the `TelePeriod` command. Then my simple test case was somewhat workable in the broken setup without networking.

**Reply by Vitek-117** [reply]
[Jason2866]
I use GPIO 5 for DS18B20 
I use GPIO 2 and 4 for TM1637.

**Comment by arendst** [comment, marked as answer]
Using different hardware (esp32c3 with W5500 ethernet cable disconnected and ``wifi 0``) I see the same situation:
```
00:00:00.050 CMD: Using USB CDC
00:00:00.051 HDW: ESP32-C3 v0.4 
00:00:00.073 UFS: FlashFS mounted with 296 kB free
00:00:00.084 CFG: Loaded from File, Count 33
00:00:00.087 SER: Set to 8N1 115200 bit/s
00:00:00.087 SER: HWCDC supports 115200 bit/s only
00:00:00.091 QPC: Count 1
00:00:00.093 SPI: Hardware using GPIO04(CLK), GPIO06(MOSI) and GPIO05(MISO)
00:00:00.096 ROT: Mode 1
00:00:00.124 CFG: No '*.autoconf' file found
00:00:00.128 BRY: Berry initialized, RAM used 3712 bytes
00:00:00.134 BRY: No 'preinit.be'
00:00:00.136 SRC: Restart
00:00:00.139 Project tasmota - Esp32c3Mini2 Version 13.4.1.2(theo)-3_0_0(2024-05-08T17:14:32)
00:00:00.171 ETH: Attempting connection...
00:00:00.195 BRY: No 'autoexec.be'
00:00:06.885 QPC: Reset
00:00:08.886 APP: Boot Count 17
```
After this the serial console is as responsive as ever. Teleperiod just won't kick in as it "makes" no sense to send tele data as there is no network active !?! With displays connected it could make sense to allow teleperiod to trigger rules for display....

This is by design as teleperiod only starts after a MQTT connection has been made.

EDIT: By executing command ``Teleperiod`` once it also starts reporting teleperiod data ready to be shown on the display. So no code change needed.
```
00:00:17.384 CMD: teleperiod
00:00:17.385 SRC: UsbConsole
00:00:17.385 CMD: Grp 0, Cmd 'TELEPERIOD', Idx 1, Len 0, Pld -99, Data ''
00:00:17.386 RSL: RESULT = {"TelePeriod":60}
00:00:17.912 RSL: STATE = {"Time":"1970-01-01T00:00:17","Uptime":"0T00:00:19","UptimeSec":19,"Heap":235,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":20,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":34},"POWER":"ON","Dimmer":100,"Color":"FFD000","HSBColor":"49,100,100","Channel":[100,82,0],"Scheme":2,"Fade":"ON","Speed":1,"LedTable":"ON"}
00:01:17.913 RSL: STATE = {"Time":"1970-01-01T00:01:17","Uptime":"0T00:01:19","UptimeSec":79,"Heap":235,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":21,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":34},"POWER":"ON","Dimmer":100,"Color":"CCFF00","HSBColor":"72,100,100","Channel":[80,100,0],"Scheme":2,"Fade":"ON","Speed":1,"LedTable":"ON"}
```

EDIT2: Adding a rule like ``on system#init do teleperiod endon`` will result in this:
```
00:00:00.051 CMD: Using USB CDC
00:00:00.051 HDW: ESP32-C3 v0.4 
00:00:00.063 UFS: FlashFS mounted with 296 kB free
00:00:00.069 CFG: Loaded from File, Count 35
00:00:00.072 SER: Set to 8N1 115200 bit/s
00:00:00.072 SER: HWCDC supports 115200 bit/s only
00:00:00.076 QPC: Count 1
00:00:00.079 SPI: Hardware using GPIO04(CLK), GPIO06(MOSI) and GPIO05(MISO)
00:00:00.081 ROT: Mode 1
00:00:00.102 CFG: No '*.autoconf' file found
00:00:00.106 BRY: Berry initialized, RAM used 3712 bytes
00:00:00.111 BRY: No 'preinit.be'
00:00:00.112 SRC: Restart
00:00:00.116 Project tasmota - Esp32c3Mini2 Version 13.4.1.2(theo)-3_0_0(2024-05-08T17:14:32)
00:00:00.149 ETH: Attempting connection...
00:00:00.164 BRY: No 'autoexec.be'
00:00:00.227 RUL: SYSTEM#INIT performs 'teleperiod'
00:00:00.227 SRC: Rule
00:00:00.228 CMD: Grp 0, Cmd 'TELEPERIOD', Idx 1, Len 0, Pld -99, Data ''
00:00:00.229 RSL: RESULT = {"TelePeriod":60}
00:00:00.976 RSL: STATE = {"Time":"1970-01-01T00:00:00","Uptime":"0T00:00:02","UptimeSec":2,"Heap":235,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":3,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":34},"POWER":"ON","Dimmer":100,"Color":"BBFF00","HSBColor":"76,100,100","Channel":[73,100,0],"Scheme":2,"Fade":"ON","Speed":1,"LedTable":"ON"}
00:00:06.903 QPC: Reset
00:00:08.900 APP: Boot Count 19
00:01:00.906 RSL: STATE = {"Time":"1970-01-01T00:01:00","Uptime":"0T00:01:02","UptimeSec":62,"Heap":235,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":24,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":34},"POWER":"ON","Dimmer":100,"Color":"B7FF00","HSBColor":"77,100,100","Channel":[72,100,0],"Scheme":2,"Fade":"ON","Speed":1,"LedTable":"ON"}
```

**Reply by Vitek-117** [reply]
[arendst]
The idea of the rules is interesting. But this is not the solution to the problem. For a good reason, the device should initially boot up and work immediately. Without any crutches in the form of rules. This is not a real solution to the problem. For example, my device on WT32-ETH01 with ESP Easy firmware immediately boots and works offline. Without any network at all. And tasmota unfortunately does not load until some kind of network appears.

**Reply by sfromis** [reply]
As you are starting Tasmota in a broken situation outside of how it intended to be used, the simple workaround should be just fine as a solution for your marginal edge case.

And again, your description does not match the facts. Tasmota **is** fully loading without network, just with the quirk of telemetry not starting after the boot completed.

**Reply by Vitek-117** [reply]
[arendst]

I tried your advice with the addition of a rule. In the console, I did this:
Teleperiod
enter
Added a rule: 
Rule1 on system#init dote period end on
Rule1 on
Result: Tasmota starts and works without a connected cable. Just as I wanted. Thanks!!!
But I don't like that I have 2 rules left to use. If I add another rule (Rule2 ON Tele-DS18B20#Temperature DO DisplayText %value%^ ENDON) Then I will have only one last available rule. Is it possible to combine two commands into one big rule? For example, like this: Rule1 on system#init do teleperiod endon ON Tele-DS18B20#Temperature DO DisplayText %value%^ ENDON
Rule1 on
Will it work?

**Reply by arendst** [reply]
Yes.

You''ll discover a whole new world by experimenting without asking ;-)

**Reply by barbudor** [reply]
And reading the docs

**Reply by Vitek-117** [reply]
Please tell me how and where to add the Teleperiod command to the source code or 
 rule: on system#init do teleperiod endon So that you don't have to add it manually every time you set up new devices. Since I want to build a firmware with my own settings and a set of sensors.
 What needs to be changed in the my_user_config file.h so that the Teleperiod function is always enabled by default?

**Comment by Jason2866** [comment]
(re)start of the WT32-ETH01 without Ethernet cable connected
```
00:00:00.001 HDW: ESP32-D0WD v1.0 
00:00:00.039 UFS: FlashFS mounted with 304 kB free
00:00:00.059 CFG: Loaded from File, Count 65
00:00:00.101 BRY: Berry initialized, RAM used 3427 bytes
00:00:00.111 Project sonoff - Tasmota Version 14.1.0.2(UNIVERSAL)-3_0_1(2024-06-14T13:06:05)
00:00:12.000 WIF: Connecting to AP1 Jason_Home_WLAN Channel 11 BSSId 00:A0:57:2A:BD:19 in mode HT20 as sonoff-DA0F94-3988...
00:00:14.162 WIF: Connected
13:14:52.278 HTP: Web server active on sonoff-DA0F94-3988 with IP address 192.168.2.158
13:14:53.471 MQT: Attempting connection...
13:14:53.488 MQT: Connected
13:14:53.491 MQT: tele/sonoff-DA0F94/LWT = Online (retained)
13:14:53.492 MQT: cmnd/sonoff-DA0F94/POWER = 
13:14:53.496 MQT: tele/sonoff-DA0F94/INFO1 = {"Info1":{"Module":"WT32_ETH01","Version":"14.1.0.2(UNIVERSAL)","FallbackTopic":"cmnd/sonoff-DA0F94_fb/","GroupTopic":"cmnd/sonoffs/"}}
13:14:53.509 MQT: tele/sonoff-DA0F94/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"sonoff-DA0F94-3988","IPAddress":"192.168.2.158","IP6Global":"fd4d:d9a8:7683:0:c6de:e2ff:feda:f94","IP6Local":"fe80::c6de:e2ff:feda:f94%st3"}}
13:14:53.532 MQT: tele/sonoff-DA0F94/INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":36}}
13:14:58.175 MQT: tele/sonoff-DA0F94/STATE = {"Time":"2024-06-14T13:14:58","Uptime":"0T00:00:17","UptimeSec":17,"Heap":147,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":99,"MqttCount":1,"Berry":{"HeapUsed":3,"Objects":37},"Wifi":{"AP":1,"SSId":"Jason_Home_WLAN","BSSId":"00:A0:57:2A:BD:19","Channel":11,"Mode":"HT20","RSSI":100,"Signal":-47,"LinkCount":1,"Downtime":"0T00:00:10"}}
```
and with Ethernet cable connected at (re)start
```
00:00:00.001 HDW: ESP32-D0WD v1.0 
00:00:00.011 UFS: FlashFS mounted with 304 kB free
00:00:00.019 CFG: Loaded from File, Count 66
00:00:00.043 BRY: Berry initialized, RAM used 3427 bytes
00:00:00.050 Project sonoff - Tasmota Version 14.1.0.2(UNIVERSAL)-3_0_1(2024-06-14T13:06:05)
00:00:01.855 ETH: Connected at 100Mbps Full Duplex, Mac C4:DE:E2:DA:0F:97, Hostname sonoff-DA0F94-3988-eth
00:00:03.426 HTP: Web server active on sonoff-DA0F94-3988-eth with IP address 192.168.2.161
13:16:38.420 MQT: Attempting connection...
13:16:38.428 MQT: Connected
13:16:38.429 MQT: tele/sonoff-DA0F94/LWT = Online (retained)
13:16:38.430 MQT: cmnd/sonoff-DA0F94/POWER = 
13:16:38.434 MQT: tele/sonoff-DA0F94/INFO1 = {"Info1":{"Module":"WT32_ETH01","Version":"14.1.0.2(UNIVERSAL)","FallbackTopic":"cmnd/sonoff-DA0F94_fb/","GroupTopic":"cmnd/sonoffs/"}}
13:16:38.456 MQT: tele/sonoff-DA0F94/INFO2 = {"Info2":{"WebServerMode":"Admin","Ethernet":{"Hostname":"sonoff-DA0F94-3988-eth","IPAddress":"192.168.2.161"}}}
13:16:38.469 MQT: tele/sonoff-DA0F94/INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":36}}
13:16:42.967 MQT: tele/sonoff-DA0F94/STATE = {"Time":"2024-06-14T13:16:42","Uptime":"0T00:00:08","UptimeSec":8,"Heap":149,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":15,"MqttCount":1,"Berry":{"HeapUsed":3,"Objects":37}}
13:16:43.354 WIF: Connecting to AP1 Jason_Home_WLAN Channel 11 BSSId 00:A0:57:2A:BD:19 in mode HT20 as sonoff-DA0F94-3988...
13:16:46.115 WIF: Connected
```

**Reply by sfromis** [reply]
The case is very different when you have Wifi available. Then it does not make much difference losing Ethernet. Where it gets more tricky is when Tasmota boots without *any* networking, and is unable to get a real time. Hence timing mechanisms like the telemetry period are not kicking in. As previously suggested (now also by arendst), this part can get rolling just issuing the `TelePeriod` command.

**Reply by Vitek-117** [reply]
[Jason2866]
With a working wifi and a disconnected LAN cable.  In this case, it works fine for me too. But I need the device to boot and be able to work autonomously without any network at all !!!
[sfromis]
To get the time without any network, you can add the smallest clock module on the DS3231. I did so.)))

**Comment by Jason2866** [comment]
For finding your hardware issue, i would disconnect the Display and the sensor. Replace the power supply. 
There is NO bug in the firmware.

**Reply by sfromis** [reply]
From what I see there is nothing to really suggest a hardware issue, and no "bug" in the Tasmota firmware either, unless you expect the telemetry period to operate without networking, time of day, and without non-local users of telemetry.

**Reply by Vitek-117** [reply]
[Jason2866]
There is a serious flaw and flaw in the firmware. And I have already described it above!!!
I have assembled a smart switch for a smart home based on WT32-ETH01 for 3 light control channels. With a Ds18b20 temperature sensor and its output to the TM1637. That's why I can't turn them off. I immediately solved the problem with time by installing the clock module on the DS3231. The power supply is very good and reliable.

**Reply by sfromis** [reply]
Well, both me and arendst (in more detail) have **already** told you how you can handle your issue, if you care to read the responses given. Very simple to do, and by far not some sort of "serious flaw". Just a small quirk.

**Reply by Vitek-117** [reply]
[sfromis]
I'll try to do what you and arendst told me how to solve the problem. 
What you say is "a small feature" Here I strongly disagree with you. This is just a serious firmware flaw and a developer's omission. Because the firmware should work regardless of the presence of any network. For example, in a field where there is nothing at all.
 I hope that the tasmota developer will eliminate this shortcoming. Like ESP Easy, for example, it generally works independently of any network.

**Reply by Jason2866** [reply]
The "m" in Tasmota stands for mqtt. It is designed to have a connection to a mqtt broker. If you want a offline solution Tasmota is the wrong choice.
